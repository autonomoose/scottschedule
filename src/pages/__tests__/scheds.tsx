import React from "react";
import { render, waitFor } from "@testing-library/react";
import userEvent from '@testing-library/user-event'

import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import List from '@mui/material/List';
import ListItem from '@mui/material/ListItem';

import { ManSched, CreateGroup, ModifyGroup } from '../../components/schedgrputil';
import SchedsPage from "../scheds";

interface mockCreateGroupProps {
  onComplete: (status: string) => void,
  open: boolean
}
export const mockCreateGroup = (props: mockCreateGroupProps) => {
    return(
      <Box data-testid='newGroup' display={(props.open)?'block': 'none'}>
      <Button onClick={() => props.onComplete('newGroupName')}>Save New</Button>
      <Button onClick={() => props.onComplete('')}>Cancel New</Button>
      </Box>
)};
interface mockModifyGroupProps {
  group: string,
  groupSched: iSchedGroup,
  onComplete: (status: string) => void,
  open: boolean,
}
export const mockModifyGroup = (props: mockModifyGroupProps) => {
    return(
      <Box data-testid='modifyGroup' display={(props.open)?'block': 'none'}>
      <Button onClick={() => props.onComplete(props.group)}>Save Mod</Button>
      <Button onClick={() => props.onComplete('')}>Cancel Mod</Button>
      <List disablePadding dense sx={{marginLeft: '1em'}}>
        <ListItem button key='testsched' onClick={() => props.onComplete('_testgrp!testsched')}>
            testsched - test sched
        </ListItem>
      </List>

      </Box>
)};

interface mockManSchedProps {
  groupSchedName: string, // group!sched or group!_NEW_
  gSchedule: iSchedule,
  onComplete: (status: string) => void,
  open: boolean
}
export const mockManSched = (props: mockManSchedProps) => {
    return(
      <Box data-testid='manSched' display={(props.open)?'block': 'none'}>
      <Button onClick={() => props.onComplete(props.groupSchedName)}>Save Sched</Button>
      <Button onClick={() => props.onComplete('')}>Cancel Sched</Button>
      </Box>
)};

jest.mock('../../components/schedgrputil', () => ({
    ...jest.requireActual('../../components/schedgrputil'),
    fetchSchedGroupsDB: jest.fn(() => Promise.resolve(
      {testgrp: {descr: 'test group',
        schedNames: [{
          begins: '8:00',
          buttonName: ' ',
          descr: 'test sched',
          schedName: 'testsched',
          schedTasks: [{evTaskId: 'testev'}],
          }]}
      })),
    ManSched: jest.fn(),
    CreateGroup: jest.fn(),
    ModifyGroup: jest.fn(),
}));

beforeAll(() => {
  (CreateGroup as jest.Mock).mockImplementation(mockCreateGroup);
  (ModifyGroup as jest.Mock).mockImplementation(mockModifyGroup);
  (ManSched as jest.Mock).mockImplementation(mockManSched);
});

const mockEnqueue = jest.fn();
jest.mock('notistack', () => ({
  ...jest.requireActual('notistack'),
  useSnackbar: () => {
    return {
      enqueueSnackbar: mockEnqueue
    };
  }
}));

const mytest = <SchedsPage />;
const mySetup = async () => {
    const utils = render(mytest);
    await waitFor(() => {
        expect(utils.getByTestId('dataBackdrop')).not.toBeVisible();
    });

    const newGroupButton = utils.getByTestId('create-group');
    const newGroupPanel = utils.getByTestId('newGroup');
    const modGroupPanel = utils.getByTestId('modifyGroup');
    const manSchedPanel = utils.getByTestId('manSched');
    return {
        ...utils,
        newGroupButton,
        newGroupPanel,
        modGroupPanel,
        manSchedPanel,
    }
}

describe("SchedsPage", () => {
  it("renders snapshot correctly", async () => {
    const {asFragment, getByTestId} = render(mytest);
    await waitFor(() => {
        expect(getByTestId('dataBackdrop')).not.toBeVisible();
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it("handles panel setup correctly", async () => {
    const utils = await mySetup();

    expect(utils.newGroupButton).toBeEnabled();
    expect(utils.newGroupPanel).not.toBeVisible();
    expect(utils.modGroupPanel).not.toBeVisible();
    expect(utils.manSchedPanel).not.toBeVisible();

  });

  it("opens new group panel", async () => {
    const utils = await mySetup();

    expect(utils.newGroupButton).toBeEnabled();
    await userEvent.click(utils.newGroupButton);
    await waitFor(() => {
      expect(utils.newGroupPanel).toBeVisible();
    });
    expect(utils.modGroupPanel).not.toBeVisible();
    expect(utils.manSchedPanel).not.toBeVisible();

  });

  it("handles new group cancel", async () => {
    const utils = await mySetup();

    await userEvent.click(utils.newGroupButton);
    await waitFor(() => {
      expect(utils.newGroupPanel).toBeVisible();
    });
    const cancelButton = utils.getByRole('button', {name: /cancel new/i});
    await userEvent.click(cancelButton);
    await waitFor(() => {
      expect(utils.newGroupPanel).not.toBeVisible();
    });

  });

  it("handles new group save", async () => {
    const utils = await mySetup();

    await userEvent.click(utils.newGroupButton);
    await waitFor(() => {
      expect(utils.newGroupPanel).toBeVisible();
    });

    const saveButton = utils.getByRole('button', {name: /save new/i});
    await userEvent.click(saveButton);
    await waitFor(() => {
      expect(utils.getByTestId('dataBackdrop')).not.toBeVisible();
    });

  });

  it("opens modify group panel", async () => {
    const utils = await mySetup();

    const modGroupButton = utils.getByRole('button', {name: /testgrp - test group/i});
    await userEvent.click(modGroupButton);
    await waitFor(() => {
      expect(utils.modGroupPanel).toBeVisible();
    });
    expect(utils.newGroupPanel).not.toBeVisible();
    expect(utils.manSchedPanel).not.toBeVisible();

  });

  it("handles modify group cancel", async () => {
    const utils = await mySetup();

    const modGroupButton = utils.getByRole('button', {name: /testgrp - test group/i});
    await userEvent.click(modGroupButton);
    await waitFor(() => {
      expect(utils.modGroupPanel).toBeVisible();
    });

    const cancelButton = utils.getByRole('button', {name: /cancel mod/i});
    await userEvent.click(cancelButton);
    await waitFor(() => {
      expect(utils.modGroupPanel).not.toBeVisible();
    });

  });

  it("handles modify group save", async () => {
    const utils = await mySetup();

    const modGroupButton = utils.getByRole('button', {name: /testgrp - test group/i});
    await userEvent.click(modGroupButton);
    await waitFor(() => {
      expect(utils.modGroupPanel).toBeVisible();
    });

    const saveButton = utils.getByRole('button', {name: /save mod/i});
    await userEvent.click(saveButton);
    await waitFor(() => {
      expect(utils.getByTestId('dataBackdrop')).not.toBeVisible();
    });
  });


  it("opens modify schedule panel and handles cancel", async () => {
    const utils = await mySetup();

    // open group
    const modGroupButton = utils.getByRole('button', {name: /testgrp - test group/i});
    await userEvent.click(modGroupButton);
    await waitFor(() => {
      expect(utils.modGroupPanel).toBeVisible();
    });

    // open sched
    const modSchedButton = utils.getByRole('button', {name: /testsched - test sched/i});
    await userEvent.click(modSchedButton);
    await waitFor(() => {
      expect(utils.manSchedPanel).toBeVisible();
    });
    expect(utils.newGroupPanel).not.toBeVisible();
    expect(utils.modGroupPanel).not.toBeVisible();

    // cancel sched
    const cancelButton = utils.getByRole('button', {name: /cancel sched/i});
    await userEvent.click(cancelButton);
    await waitFor(() => {
      expect(utils.manSchedPanel).not.toBeVisible();
    });

    // should show group list
    expect(modGroupButton).toBeVisible();
  });

  it("handles modify schedule save", async () => {
    const utils = await mySetup();

    // open group
    const modGroupButton = utils.getByRole('button', {name: /testgrp - test group/i});
    await userEvent.click(modGroupButton);
    await waitFor(() => {
      expect(utils.modGroupPanel).toBeVisible();
    });

    // open sched
    const modSchedButton = utils.getByRole('button', {name: /testsched - test sched/i});
    await userEvent.click(modSchedButton);
    await waitFor(() => {
      expect(utils.manSchedPanel).toBeVisible();
    });

    const saveButton = utils.getByRole('button', {name: /save sched/i});
    await userEvent.click(saveButton);
    await waitFor(() => {
      expect(utils.getByTestId('dataBackdrop')).not.toBeVisible();
    });

    expect(utils.manSchedPanel).toBeVisible();
  });

});
