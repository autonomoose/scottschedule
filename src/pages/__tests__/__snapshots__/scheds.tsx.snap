// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchedsPage renders snapshot correctly 1`] = `
<div
  data-testid="authenticator"
>
   
  <div
    style="margin: 1rem auto; min-height: 100vh; text-align: center;"
  >
    <div
      class="MuiBox-root css-0"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorTransparent MuiAppBar-positionFixed mui-fixed css-1bzzxac-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1ti0wu3-MuiToolbar-root"
        >
          <button
            aria-label="Open menu"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1t7w49f-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
            href="/home"
          >
            <div
              class="MuiBox-root css-k008qs"
            >
              <img
                alt="Werner Digital"
                src="../images/wernerdigital-hosted.png"
                width="26"
              />
              <p
                class="MuiTypography-root MuiTypography-body1 css-4peds9-MuiTypography-root"
              >
                Scottschedule
              </p>
            </div>
          </a>
          <div
            class="MuiBox-root css-yukqif"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
              href="/usermaint"
              style="margin-right: 1rem;"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                mockuname
              </p>
            </a>
            <button
              class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1vbvugf-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              Sign Out
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
          <button
            aria-label="Dark/light mode"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeStart MuiIconButton-sizeMedium css-1t7w49f-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="LightModeIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 7c-2.76 0-5 2.24-5 5s2.24 5 5 5 5-2.24 5-5-2.24-5-5-5zM2 13h2c.55 0 1-.45 1-1s-.45-1-1-1H2c-.55 0-1 .45-1 1s.45 1 1 1zm18 0h2c.55 0 1-.45 1-1s-.45-1-1-1h-2c-.55 0-1 .45-1 1s.45 1 1 1zM11 2v2c0 .55.45 1 1 1s1-.45 1-1V2c0-.55-.45-1-1-1s-1 .45-1 1zm0 18v2c0 .55.45 1 1 1s1-.45 1-1v-2c0-.55-.45-1-1-1s-1 .45-1 1zM5.99 4.58c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0s.39-1.03 0-1.41L5.99 4.58zm12.37 12.37c-.39-.39-1.03-.39-1.41 0-.39.39-.39 1.03 0 1.41l1.06 1.06c.39.39 1.03.39 1.41 0 .39-.39.39-1.03 0-1.41l-1.06-1.06zm1.06-10.96c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06zM7.05 18.36c.39-.39.39-1.03 0-1.41-.39-.39-1.03-.39-1.41 0l-1.06 1.06c-.39.39-.39 1.03 0 1.41s1.03.39 1.41 0l1.06-1.06z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </header>
      <div
        class="MuiDrawer-root MuiDrawer-docked css-7ik032-MuiDrawer-docked"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-1hrb81m-MuiPaper-root-MuiDrawer-paper"
          style="visibility: hidden; webkit-transform: translateX(-0px); transform: translateX(-0px);"
        >
          <div
            class="MuiBox-root css-1ecnymn"
          >
            <div
              class="MuiBox-root css-xwg089"
            >
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
              >
                MainÂ Menu
              </h6>
              <button
                aria-label="Close menu"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1drgtl0-MuiButtonBase-root-MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="ArrowDropUpIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m7 14 5-5 5 5z"
                  />
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-rr7dug-MuiDivider-root"
          />
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
              href="/home"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="HomeIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Home
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
              href="/usermaint"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Account
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
              href="/scheds"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Schedules
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
              href="/events"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Events
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-14b5yu0-MuiTypography-root-MuiLink-root"
              href="/help"
            >
              <div
                class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
                role="button"
                tabindex="0"
              >
                <div
                  class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="StarIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <span
                    class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                  >
                    Help
                  </span>
                </div>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </div>
            </a>
            <div
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-ltlrkn-MuiButtonBase-root-MuiListItem-root"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root css-qklaqr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="StarIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 17.27 18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  Logout
                </span>
              </div>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </div>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-h8m1q6"
    >
      <main>
        <div
          data-testid="mainPageDisplay"
          style="text-align: left;"
        >
          <div
            class="MuiBox-root css-7kebnq"
          >
            <div
              style="float: left;"
            >
              <nav
                aria-label="Breadcrumb"
                class="MuiTypography-root MuiTypography-body1 MuiBreadcrumbs-root css-1kkz7jp-MuiTypography-root-MuiBreadcrumbs-root"
              >
                <ol
                  class="MuiBreadcrumbs-ol css-4pdmu4-MuiBreadcrumbs-ol"
                >
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <a
                      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
                      href="/home"
                    >
                      Home
                    </a>
                  </li>
                  <li
                    aria-hidden="true"
                    class="MuiBreadcrumbs-separator css-1wuw8dw-MuiBreadcrumbs-separator"
                  >
                    &gt;
                  </li>
                  <li
                    class="MuiBreadcrumbs-li"
                  >
                    <span>
                      Schedules
                    </span>
                  </li>
                </ol>
              </nav>
            </div>
            <div>
              <a
                class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
                href="/help/scheds"
              >
                Help
              </a>
            </div>
          </div>
          <div
            class="MuiBox-root css-7m4n47"
          >
            <div
              class="MuiBox-root css-0"
            >
              <div
                class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1gyd2pn-MuiPaper-root-MuiCard-root"
                style="max-width: 432px; min-width: 394px; flex: 1 1; box-shadow: 5px 5px 12px #888888; border-radius: 0 0 5px 5px;"
              >
                <h6
                  class="MuiTypography-root MuiTypography-h6 css-1eaq3q5-MuiTypography-root"
                >
                  Schedule and Schedule Groups Editor
                </h6>
                <p
                  class="MuiTypography-root MuiTypography-body2 css-17f9bli-MuiTypography-root"
                >
                  Create and modify schedules, and organize them into groups
                </p>
                <div
                  class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAccordion-root MuiAccordion-rounded Mui-expanded css-vrg8eu-MuiPaper-root-MuiAccordion-root"
                >
                  <div
                    aria-expanded="true"
                    class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded css-1ruslfs-MuiButtonBase-root-MuiAccordionSummary-root"
                    role="button"
                    tabindex="0"
                  >
                    <div
                      class="MuiAccordionSummary-content Mui-expanded css-1betqn-MuiAccordionSummary-content"
                    >
                      <div
                        class="MuiBox-root css-pvuze6"
                      >
                        Schedule Groups List (
                        1
                        )
                        <button
                          class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1vbvugf-MuiButtonBase-root-MuiButton-root"
                          data-testid="create-group"
                          tabindex="0"
                          type="button"
                        >
                          New Group
                        </button>
                      </div>
                    </div>
                    <div
                      class="MuiAccordionSummary-expandIconWrapper Mui-expanded css-19hn6xk-MuiAccordionSummary-expandIconWrapper"
                    >
                      <svg
                        aria-hidden="true"
                        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="ExpandMoreIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                        />
                      </svg>
                    </div>
                  </div>
                  <div
                    class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
                    style="min-height: 0px;"
                  >
                    <div
                      class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
                    >
                      <div
                        class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
                      >
                        <div
                          class="MuiAccordion-region"
                          role="region"
                        >
                          <div
                            class="MuiAccordionDetails-root css-62cn81-MuiAccordionDetails-root"
                          >
                            <ul
                              class="MuiList-root MuiList-dense css-16vbaq4-MuiList-root"
                            >
                              <div
                                class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding MuiListItem-button css-1cnxyiq-MuiButtonBase-root-MuiListItem-root"
                                role="button"
                                tabindex="0"
                              >
                                testgrp
                                 - 
                                test group
                              </div>
                            </ul>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiBox-root css-1hyfx7x"
              data-testid="newGroup"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1o011v7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Save New
              </button>
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1o011v7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Cancel New
              </button>
            </div>
            <div
              class="MuiBox-root css-1hyfx7x"
              data-testid="modifyGroup"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1o011v7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Save Mod
              </button>
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1o011v7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Cancel Mod
              </button>
              <ul
                class="MuiList-root MuiList-dense css-16vbaq4-MuiList-root"
              >
                <div
                  class="MuiButtonBase-root MuiListItem-root MuiListItem-dense MuiListItem-gutters MuiListItem-padding MuiListItem-button css-1cnxyiq-MuiButtonBase-root-MuiListItem-root"
                  role="button"
                  tabindex="0"
                >
                  testsched - test sched
                </div>
              </ul>
            </div>
            <div
              class="MuiBox-root css-1hyfx7x"
              data-testid="manSched"
            >
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1o011v7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Save Sched
              </button>
              <button
                class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1o011v7-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                Cancel Sched
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            class="MuiBackdrop-root css-1178cxv-MuiBackdrop-root"
            style="opacity: 0; visibility: hidden;"
          >
            <span
              class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
              data-testid="dataBackdrop"
              role="progressbar"
              style="width: 40px; height: 40px;"
            >
              <svg
                class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
                  cx="44"
                  cy="44"
                  fill="none"
                  r="20.2"
                  stroke-width="3.6"
                />
              </svg>
            </span>
          </div>
        </div>
      </main>
    </div>
    <footer
      style="padding-top: 40px;"
    >
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-rr7dug-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-1bnxa12"
      >
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
          href="/home"
        >
          Home
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
          href="/usermaint"
        >
          Account
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
          href="/scheds"
        >
          Schedules
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
          href="/events"
        >
          Events
        </a>
        <a
          class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-yddyga-MuiTypography-root-MuiLink-root"
          href="/help"
        >
          Help
        </a>
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-rr7dug-MuiDivider-root"
      />
      <span
        class="MuiTypography-root MuiTypography-caption css-3merus-MuiTypography-root"
      >
        Â© 2021-
        2022
        , Werner Digital Technology Inc
      </span>
    </footer>
  </div>
</div>
`;
